lms/djangoapps/courseware/tests/test_tabs.py

```diff
@@ -170,3 +170,50 @@ def test_get_course_tabs_list_entrance_exam_enabled(self):
             self.assertEqual(course_tab_list[0]['tab_id'], 'courseware')
             self.assertEqual(course_tab_list[0]['name'], 'Entrance Exam')
             self.assertEqual(course_tab_list[1]['tab_id'], 'instructor')
+
+
+@override_settings(MODULESTORE=TEST_DATA_MIXED_TOY_MODULESTORE)
+class TextBookTabsTestCase(LoginEnrollmentTestCase, ModuleStoreTestCase):
+    """
+    Validate tab behavior when dealing with textbooks.
+    """
+
+    def setUp(self):
+        self.course = CourseFactory.create()
+        self.set_up_books(2)
+        self.course.tabs = [
+            tabs.CoursewareTab(),
+            tabs.CourseInfoTab(),
+            tabs.TextbookTabs(),
+            tabs.PDFTextbookTabs(),
+            tabs.HtmlTextbookTabs(),
+        ]
+        self.setup_user()
+        self.enroll(self.course)
+
+    def set_up_books(self, num_books):
+        """Initializes the textbooks in the course and adds the given number of books to each textbook"""
+        self.books = [MagicMock() for _ in range(num_books)]
+        for book_index, book in enumerate(self.books):
+            book.title = 'Book{0}'.format(book_index)
+        self.course.textbooks = self.books
+        self.course.pdf_textbooks = self.books
+        self.course.html_textbooks = self.books
+
+    def test_pdf_textbook_tabs(self):
+        """
+        Test that all textbooks tab links generating correctly.
+        """
+        type_to_reverse_name = {'textbook': 'book', 'pdftextbook': 'pdf_book', 'htmltextbook': 'html_book'}
+
+        course_tab_list = get_course_tab_list(self.course, self.user)
```

Comment:
you might want to assert that there are three `SingleTextbookTabs` in `course_tab_list`